File: _tools/scratch/generate_monster_report.py
import os
import json
import glob

monsters_dir = r'e:\AntigravitiProgetti\CompendioTheWitcher\_tools\src-packs\BESTIARIO\witcher-monsters'
output_file = r'e:\AntigravitiProgetti\CompendioTheWitcher\TO DO\report_mostri.md'

files = glob.glob(os.path.join(monsters_dir, '*.json'))

monsters = []

for file in files:
    with open(file, 'r', encoding='utf-8') as f:
        try:
            data = json.load(f)
            name = data.get('name', 'Sconosciuto')
            system = data.get('system', {})
            
            # Stats
            stats = system.get('stats', {})
            stats_present = any(s.get('value', 0) > 0 for s in stats.values() if isinstance(s, dict))
            
            # Derived
            derived = system.get('derivedStats', {})
            derived_present = any(s.get('value', 0) > 0 for s in derived.values() if isinstance(s, dict))
            
            # Skills
            skills = system.get('skills', {})
            skill_list = []
            for stat_key, stat_skills in skills.items():
                if isinstance(stat_skills, dict):
                    for sk_name, sk_data in stat_skills.items():
                        if isinstance(sk_data, dict) and sk_data.get('value', 0) > 0:
                            skill_list.append(f"{sk_name} ({sk_data.get('value')})")
            
            # Items
            items = data.get('items', [])
            weapons = []
            abilities = []
            for item in items:
                item_type = item.get('type')
                item_name = item.get('name', '')
                if item_type == 'weapon':
                    weapons.append(item_name)
                elif item_type == 'ability':
                    abilities.append(item_name)
            
            monsters.append({
                'name': name,
                'stats': '✅' if stats_present else '❌',
                'derived': '✅' if derived_present else '❌',
                'skills': len(skill_list),
                'weapons': ', '.join(weapons) if weapons else '-',
                'abilities': ', '.join(abilities) if abilities else '-'
            })
        except Exception as e:
            print(f"Error processing {file}: {e}")

monsters.sort(key=lambda x: x['name'])

with open(output_file, 'w', encoding='utf-8') as f:
    f.write('# Report Mostri\n\n')
    f.write('| Mostro | Statistiche | Stat Derivate | N° Abilità (Skills) | Armi/Attacchi | Capacità (Abilities) |\n')
    f.write('|---|---|---|---|---|---|\n')
    for m in monsters:
        f.write(f"| **{m['name']}** | {m['stats']} | {m['derived']} | {m['skills']} | {m['weapons']} | {m['abilities']} |\n")

print(f"Report generato in {output_file}")

File: _tools/_garbage/scratch/generate_monster_report.js
const fs = require('fs');
const path = require('path');

const monstersDir = path.join(__dirname, '..', 'src-packs', 'BESTIARIO', 'witcher-monsters');
const outputFile = path.join(__dirname, '..', '..', 'TO DO', 'report_mostri.md');

const files = fs.readdirSync(monstersDir).filter(f => f.endsWith('.json'));

const monsters = [];

for (const file of files) {
    const filePath = path.join(monstersDir, file);
    try {
        const content = fs.readFileSync(filePath, 'utf-8');
        const data = JSON.parse(content);
        
        const name = data.name || 'Sconosciuto';
        const system = data.system || {};
        
        // Stats
        const stats = system.stats || {};
        let statsPresent = false;
        for (const key in stats) {
            if (stats[key] && stats[key].value > 0) {
                statsPresent = true;
                break;
            }
        }
        
        // Derived
        const derived = system.derivedStats || {};
        let derivedPresent = false;
        for (const key in derived) {
            if (derived[key] && derived[key].value > 0) {
                derivedPresent = true;
                break;
            }
        }
        
        // Skills
        const skills = system.skills || {};
        let skillCount = 0;
        for (const statKey in skills) {
            const statSkills = skills[statKey];
            if (typeof statSkills === 'object') {
                for (const skName in statSkills) {
                    if (statSkills[skName] && statSkills[skName].value > 0) {
                        skillCount++;
                    }
                }
            }
        }
        
        // Items
        const items = data.items || [];
        const weapons = [];
        const abilities = [];
        for (const item of items) {
            if (item.type === 'weapon') {
                weapons.push(item.name);
            } else if (item.type === 'ability') {
                abilities.push(item.name);
            }
        }
        
        monsters.push({
            name,
            stats: statsPresent ? '✅' : '❌',
            derived: derivedPresent ? '✅' : '❌',
            skills: skillCount,
            weapons: weapons.length > 0 ? weapons.join(', ') : '-',
            abilities: abilities.length > 0 ? abilities.join(', ') : '-'
        });
    } catch (e) {
        console.error(`Error processing ${file}:`, e);
    }
}

monsters.sort((a, b) => a.name.localeCompare(b.name));

let markdown = '# Report Mostri\n\n';
markdown += '| Mostro | Statistiche | Stat Derivate | N° Abilità (Skills) | Armi/Attacchi | Capacità (Abilities) |\n';
markdown += '|---|---|---|---|---|---|\n';
for (const m of monsters) {
    markdown += `| **${m.name}** | ${m.stats} | ${m.derived} | ${m.skills} | ${m.weapons} | ${m.abilities} |\n`;
}

fs.writeFileSync(outputFile, markdown, 'utf-8');
console.log(`Report generato in ${outputFile}`);
